docs/tutorial/az-mysql-tutorial.md

```diff
@@ -0,0 +1,531 @@
+- Set up primary and secondary MySQL databases on the VMs
+- Orchestrate master-replica replication across the two zones with SSL encryption
+- Configure automatic failover
+- Test and validate the recovery
+
+By the end of this tutorial, you will have a working high availability MySQL environment capable of promoting replicas, reintegrating failed nodes, and maintaining service continuity entirely within a local setup.
+
+``` {important}
+The process has been validated using Multipass 1.17 and MySQL 8.0.
+```
+
+## 1. Set up the VMs
+
+Begin by launching two virtual machines, `mysql-primary` and `mysql-standby`, and distributing them across different zones (in this case zone1 and zone2) to simulate distinct availability zones:
+
+```bash
+multipass launch --name mysql-primary --zone zone1
+multipass launch --name mysql-standby --zone zone2
+```
+
+After launching, install MySQL Server on both VMs:
+
+```bash
+for vm in mysql-primary mysql-standby; do
+  multipass exec $vm -- sudo apt update
+  multipass exec $vm -- sudo apt install -y mysql-server
+done
+```
+
+This installation process ensures necessary packages are in place.
+
+## 2. Configure the primary MySQL server
+
+Access the primary VM, `mysql-primary`, to begin its configuration:
+
+```bash
+multipass shell mysql-primary
+```
+
+First, configure MySQL authentication for replication. Since replication requires network connections, we need to set up password authentication:
+
+```bash
+# Connect to MySQL as root using sudo (auth_socket)
+sudo mysql << 'EOF'
+-- Set a password for root user and enable password authentication
+ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY 'SecureRootPass123';
+-- Allow root connections from any host (needed for replication)
+CREATE USER 'root'@'%' IDENTIFIED BY 'SecureRootPass123';
+GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' WITH GRANT OPTION;
+-- Remove anonymous users and test database for security
+DROP USER IF EXISTS ''@'localhost';
+DROP USER IF EXISTS ''@'%';
+DROP DATABASE IF EXISTS test;
+FLUSH PRIVILEGES;
+EOF
+```
+
+**Important**: Replace `SecureRootPass123` with a secure password of your choice throughout this guide.
+
+Now configure MySQL for replication. Open the MySQL configuration file:
+
+```bash
+sudo nano /etc/mysql/mysql.conf.d/mysqld.cnf
+```
+
+Add these settings to the `[mysqld]` section:
+
+```ini
+server-id=1
+log_bin=mysql-bin
+binlog_do_db=testdb
+bind-address=0.0.0.0
+
+# SSL configuration for secure replication
+ssl-ca=/etc/mysql/ssl/ca-cert.pem
+ssl-cert=/etc/mysql/ssl/server-cert.pem
+ssl-key=/etc/mysql/ssl/server-key.pem
+```
+
+Restart MySQL after applying these changes:
+
+```bash
+sudo systemctl restart mysql
+```
+
+Verify MySQL is listening on all interfaces:
+
+```bash
+sudo ss -tlnp | grep :3306
+```
+
+You should see MySQL listening on `0.0.0.0:3306`. If you see `127.0.0.1:3306`, check that `bind-address=0.0.0.0` is properly set in the configuration file.
```

Comment:
The default configuration also includes a `mysqlx-bind-address` which when not overridden with `0.0.0.0` results in the server listening on `127.0.0.1:33060` which makes this line confusing. We should either tighten the `grep` command or also override this `bind-address` as well.